source/manuals/programming-languages/html.html.md.erb

```diff
@@ -1,85 +1,228 @@
+## Element structure
 
-The following roles map directly to HTML5 sectioning elements:
+### Self-closing tags
 
-* `role="main"` maps to the `<main>` element
-* `role="complementary"` maps to the `<aside>` element
-* `role="contentinfo"` maps to the `<footer>` element
-* `role="navigation"` maps to the `<nav>` element
+[End slashes are optional in HTML5](https://www.w3.org/TR/html5/syntax.html#optional-start-and-end-tags),
+meaning it is possible to write `<br>` as well as `<br />`. However, without
+web developer experience, it is impossible to know whether an element is a
+[void (self-closing)](https://www.w3.org/TR/html5/syntax.html#void-elements)
+element (e.g. `<br>`) or a container element (e.g. `<div></div>`). If you
+encounter a custom element in the DOM, e.g. `<foo>`, do you know if it is
+implicitly closed or should you be looking for a closing `</foo>` tag?
 
-[This article from Léonie Watson](http://tink.uk/screen-readers-aria-roles-html5-support/) explains why using ARIA landmark roles in this way delivers the best experience for screen reader users.
+For consistency, readability and a lower learning curve for new developers,
+we suggest *always explicitly closing void elements*.
 
-For a more comprehensive guide to the mapping of roles to HTML5 sectioning elements read this [guide](http://blog.paciellogroup.com/2013/02/using-wai-aria-landmarks-2013/) from the Paciello Group.
+```html
+<!-- Good -->
+<br />
+<img src="foo.jpg" alt="example" />
+<p>This is a paragraph.</p>
 
-The following landmark roles don't map to HTML5 elements, but are
-still good to use:
+<!-- Bad -->
+<br>
+<img src="foo.jpg" alt="example">
+<p>Notice no closing tag.
+<img src="bar.jpg" alt="another example"></img>
+```
 
-* `role="search"`
-* `role="application"`
+### Attributes
 
-[Demo of how landmark roles help screen reader users](http://tink.co.uk/2011/07/how-do-aria-landmark-roles-help-screen-reader-users/)
+Always quote attribute values. Quotes should be double-quoted (`"`) rather
+than single-quoted (`'`) unless you're rendering HTML through template strings
+(e.g. in JavaScript), in which case either single or double quotes is fine.
 
-## Main content
+Some attributes don't require values. In these cases, omit them.
 
-The `<main>` tag should be used to identify the main content of your page. Support for it is still marginal so ensure to add the `role="main"` attribute.
+```html
+<!-- Bad -->
+<img height=30 width=30 src="foo.png" alt="logo" />
+<input type="checkbox" name="vehicle3" value="Boat" checked="checked" />
 
-## Sectioned content
+<!-- Good -->
+<img height="30" width="30" src="foo.png" alt="logo" />
+<input type="checkbox" name="vehicle3" value="Boat" checked />
+```
 
-When using the `<section>` tag, add an `aria-labelledby` attribute linking to the id of the heading that identifies that section.
+## Document structure
 
-Example:
+We use HTML5 to structure our page semantically. Assistive technology support
```

Comment:
I think this statement is sweeping and should be amended; I'd suggest an alternative wording, since support isn't [marginal any more](https://tink.uk/screen-reader-support-for-html5-sections/) (and has hopefully improved since that article was published in 2013).

```suggestion
We use HTML5 to structure our page semantically. We also use ARIA roles to help older assistive technologies map the sections correctly.
```

It'd be interesting to run the [HTML5 Accessibility Page](http://www.html5accessibility.com/html5elements/) through the Empathy Lab to see how much has or hasn't improved.